File: sdk/vc.py
"""Verifiable Credential client for AgentTrust Protocol."""
import httpx


class VCClient:
    def __init__(self, server_url: str, agent_did: str):
        self.server_url = server_url.rstrip("/")
        self.agent_did = agent_did

    async def issue(self, credential_type: str = "AgentTrustScore", expiration_days: int = 30) -> dict:
        async with httpx.AsyncClient() as client:
            r = await client.post(
                f"{self.server_url}/vc/issue",
                json={
                    "agent_did": self.agent_did,
                    "credential_type": credential_type,
                    "expiration_days": expiration_days,
                },
            )
            r.raise_for_status()
            return r.json()

    async def verify(self, credential: dict) -> dict:
        async with httpx.AsyncClient() as client:
            r = await client.post(
                f"{self.server_url}/vc/verify",
                json={"verifiable_credential": credential},
            )
            r.raise_for_status()
            return r.json()

File: sdk-ts/src/vc.ts
export class VCClient {
  constructor(private serverUrl: string, private agentDid: string) {}

  async issue(credentialType = 'AgentTrustScore', expirationDays = 30): Promise<Record<string, unknown>> {
    const r = await fetch(`${this.serverUrl}/vc/issue`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ agent_did: this.agentDid, credential_type: credentialType, expiration_days: expirationDays }),
    });
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }

  async verify(credential: Record<string, unknown>): Promise<Record<string, unknown>> {
    const r = await fetch(`${this.serverUrl}/vc/verify`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ verifiable_credential: credential }),
    });
    if (!r.ok) throw new Error(await r.text());
    return r.json();
  }
}
